src/ui/studio/save-creation/index.tsx

```diff
@@ -415,12 +431,49 @@ const UploadForm = ({ uploadState, handleUpload }) => {
     }
   }
 
+  async function handleSelectChange(newValue: SingleValue<SeriesOption>) {
+    let newSeriesValue = newValue ? newValue.value : null;
+    dispatch({ type: 'UPDATE_SERIES', value: newSeriesValue });
+  }
+
+  const filterSeries = async(inputValue: string) => {
+    const seriesList: SeriesOption[] = [];
+    const ocSeries = await opencast.getSeries(inputValue);
+    for (const [key, value] of ocSeries ) {
+      seriesList.push({ value: key, label: value });
+      if (seriesId && seriesId === key) {
+        seriesDefault = { value: key, label: value };
+      }
+    }
+    return seriesList;
+  };
+
+  const lastTimeout = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  const loadSeriesOptions = (inputValue: string) => {
+    const seriesOptions = new Promise<SeriesOption[]>(resolve => {
+      if (lastTimeout.current !== null) {
+        clearTimeout(lastTimeout.current);
+      }
+      lastTimeout.current = setTimeout(() => {
+        resolve(filterSeries(inputValue));
+      }, 30);
+    });
+    seriesOptions.then(options => {
+      if(seriesId) setSeriesValue(options.find(option => option.value === seriesId));
+    });
+    return seriesOptions;
+  };
+
   // If the user has not yet changed the value of the field and the last used
   // presenter name is used in local storage, use that.
   useEffect(() => {
     if (presenterValue !== presenter) {
       dispatch({ type: 'UPDATE_PRESENTER', value: presenterValue });
     }
+    if (seriesId != null) {
+      setValue('series', seriesId);
+    }
```

Comment:
The comment could use adjustment or an additional sentence about the lines you added.